webknossos-datastore/app/com/scalableminds/webknossos/datastore/explore/ExploreRemoteLayerService.scala

```diff
@@ -106,26 +107,93 @@ class ExploreRemoteLayerService @Inject()(dataVaultService: DataVaultService,
         uri.getPath.startsWith(whitelistEntry))) ?~> s"Absolute path ${uri.getPath} in local file system is not in path whitelist. Consider adding it to datastore.localFolderWhitelist"
     } else Fox.successful(())
 
-  private def exploreRemoteLayersForRemotePath(remotePath: VaultPath,
-                                               credentialId: Option[String],
-                                               reportMutable: ListBuffer[String],
-                                               explorers: List[RemoteLayerExplorer])(
+  private val MAX_RECURSIVE_SEARCH_DEPTH = 8
+  private def handleExploreResult(explorationResult: Box[List[(DataLayerWithMagLocators, VoxelSize)]],
+                                  explorer: RemoteLayerExplorer,
+                                  path: VaultPath)(
+      implicit reportMutable: ListBuffer[String],
+      ec: ExecutionContext): Fox[List[(DataLayerWithMagLocators, VoxelSize)]] = explorationResult match {
+    case Full(layersWithVoxelSizes) =>
+      reportMutable += s"Found ${layersWithVoxelSizes.length} ${explorer.name} layers at $path."
+      Fox.successful(layersWithVoxelSizes)
+    case f: Failure =>
+      reportMutable += s"Error when reading $path as ${explorer.name}: ${Fox.failureChainAsString(f)}"
+      Fox.empty
+    case Empty =>
+      reportMutable += s"Error when reading $path as ${explorer.name}: Empty"
+      Fox.empty
+
+  }
+
+  private def handleExplorationResultOfPath(explorationResultOfPath: Box[List[(DataLayerWithMagLocators, VoxelSize)]],
+                                            path: VaultPath,
+                                            searchDepth: Int,
+                                            remainingPaths: List[(VaultPath, Int)],
+                                            credentialId: Option[String],
+                                            explorers: List[RemoteLayerExplorer],
+                                            reportMutable: ListBuffer[String])(implicit ec: ExecutionContext) =
+    explorationResultOfPath match {
+      case Full(layersWithVoxelSizes) =>
+        val sameParentRemainingPaths = remainingPaths.filter(_._1.parent == path.parent)
+        // Try to look for additional layers in sibling directories.
+        // TODO: Problem! This runs in parallel with other sub dirs
```

Comment:
TODO?

Reply:
I introduced this search for additional layers before we moved the explorers of more elaborate dataset formats ahead (like we moved NgffExplorer ahead of the zarr explorer) to ensure finding all missing datalayers.

But maybe this is not needed anymore:

I see the following problems here: 
1. In case we have a more elaborate dataset format listing all layers in a config file like `.zarray` one could have the following folder / nested key structure:
- top level dir
  - dataset 1
    - .zarray
  - dataset 2
    - .zarray
In that case the code looking for a sibling layer is likely to find dataset 1 and dataset 2, concat them and treat them as one dataset. This seems like a bug to me, but I would need to create a test scenario to confirm this.

2. In case we have a simple dataset format without a config file and only layers:
- top level dir
  - dataset 1
    - layer 1
    - layer 2
    - layer 3
  - dataset 2
    - some layers

In this scenario, the code looking for sibling layers is likely useful, as it finds layer 1, layer 2, layer 3 of dataset 1 in one go.  But this also needs testing. @fm3 Do you maybe know how to test this?

Thus, I'd say the sibling exploration is useful in the second case but should be avoided in the first case. Maybe we can add a boolean instance val to the explorers that mark them whether they are for simple or advanced formats and only use the sibling exploration for simple dataset format explorers?

Reply:
Yes, an `exploreSiblings=true` bool seems reasonable. I’m afraid I don’t have a test dataset. Maybe you could create one in the local file system by modifying an existing one?

I would assume the siblings of the top-level dir (the one of the uri the user originally submitted) are not searched. Is this already so?

Reply:
> I’m afraid I don’t have a test dataset. Maybe you could create one in the local file system by modifying an existing one?

Oh right. That seems like an easy way to test :)

> I would assume the siblings of the top-level dir (the one of the uri the user originally submitted) are not searched. Is this already so?

Yeah, this should be the case, but I can test this.